### ISSUE TITLE
bug: Migrating bridge node keys throws error

### ISSUE THREAD

**Issue by jim-counter** [celestiaorg/celestia-node#1932]
### Celestia Node version

v0.7.1

### OS

Ubuntu 22.04.2 LTS

### Install tools

make

### Steps to reproduce it

`init` a bridge node like this:

`celestia bridge init --core.ip localhost --p2p.network blockspacerace`

Copy the two files which are in the `.celestia-bridge-blockspacerace-0\keys` folder from the source server to the destination server. In my case the files are named `NJ3XILLTMVRXEZLUFZVHO5A` and `OAZHALLLMV4Q`. I have been told this is the way to preserve the identity of the bridge node between servers during a migration.

Then start the server like this:

`celestia bridge start --core.ip localhost --p2p.network blockspacerace --metrics.tls=false --metrics --metrics.endpoint otel.celestia.tools:4318`

### Expected result

Bridge node starts with original identity.

### Actual result

Node crashes with a panic.

### Relevant log output

```Shell
celestia bridge start --core.ip localhost --p2p.network blockspacerace --metrics.tls=false --metrics --metrics.endpoint otel.celestia.tools:4318
2023-03-19T23:57:07.746+0100    INFO    node    nodebuilder/module.go:22        Accessing keyring...
2023-03-19T23:57:07.755+0100    INFO    module/state    state/keyring.go:52     constructed keyring signer      {"backend": "test", "path": "/home/jim/.celestia-bridge-blockspacerace-0/keys", "key name": "my_celes_key", "chain-id": "blockspacerace-0"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x22090a1]

goroutine 1 [running]:
github.com/celestiaorg/celestia-node/nodebuilder.(*fsStore).Close(0xc0014d9ec0?)
        /home/jim/celestia-node/nodebuilder/store.go:161 +0x61
go.uber.org/multierr.Invoke.Invoke(0x2?)
        /home/jim/go/pkg/mod/go.uber.org/multierr@v1.9.0/error.go:580 +0x1a
go.uber.org/multierr.AppendInvoke(0xc001d61a10, {0x35c7220?, 0xc0014d85a0?})
        /home/jim/go/pkg/mod/go.uber.org/multierr@v1.9.0/error.go:661 +0x30
github.com/celestiaorg/celestia-node/cmd.Start.func1(0xc000172000, {0xc001fc3900?, 0x8?, 0x8?})
        /home/jim/celestia-node/cmd/start.go:56 +0x7a2
github.com/spf13/cobra.(*Command).execute(0xc000172000, {0xc001fc3880, 0x8, 0x8})
        /home/jim/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:916 +0x862
github.com/spf13/cobra.(*Command).ExecuteC(0x4a6fba0)
        /home/jim/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
        /home/jim/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968
github.com/spf13/cobra.(*Command).ExecuteContext(...)
        /home/jim/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:961
main.run()
        /home/jim/celestia-node/cmd/celestia/main.go:32 +0xab
main.main()
        /home/jim/celestia-node/cmd/celestia/main.go:23 +0x19
```

**Comment by renaynay** [comment]
I cannot reproduce this error with the steps you mentioned. Are you sure this is the full stack trace?

**Comment by jim-counter** [comment]
Hey @renaynay thanks for looking. Sorry for the radio silence, just wanted to let you know that I am planning on trying to reproduce with my keys on a fresh Ubuntu install to see if I can get you a nice tight definition of the problem. Bear with me.

**Comment by jim-counter** [comment]
I've just spun up a test system and think I've found the problem. I am copying the two key files into the `.celestia-bridge-blockspacerace-0/keys` folder from a repository I am pulling onto the server. On this new environment I get a different error message:

```
Error: could not build arguments for function "github.com/celestiaorg/celestia-node/nodebuilder/p2p".Listen.func1 (/home/jim/celestia-node/nodebuilder/p2p/addrs.go:13): failed to build host.Host: could not build arguments for function "github.com/celestiaorg/celestia-node/nodebuilder/p2p".routedHost (/home/jim/celestia-node/nodebuilder/p2p/host.go:26): failed to build p2p.HostBase: could not build arguments for function "github.com/celestiaorg/celestia-node/nodebuilder/p2p".host (/home/jim/celestia-node/nodebuilder/p2p/host.go:31): failed to build peer.ID: could not build arguments for function "github.com/celestiaorg/celestia-node/nodebuilder/p2p".id (/home/jim/celestia-node/nodebuilder/p2p/identity.go:45): failed to build crypto.PrivKey: received non-nil error from function "github.com/celestiaorg/celestia-node/nodebuilder/p2p".Key (/home/jim/celestia-node/nodebuilder/p2p/identity.go:17): keystore: permissions of key 'p2p-key' are too relaxed: required: 0600, got: 0664
```

This bit being the important part: `keystore: permissions of key 'p2p-key' are too relaxed: required: 0600, got: 0664`.

After `chmod`ing the two key files the bridge fires up with the expected identity.

So in conclusion, no bug here but I don't know why I was getting a different error message on my main server originally. I tested this behaviour on v0.7.0 (where I got the initial error) and 0.7.2 and both behave as per this latest update.

**Comment by jim-counter** [comment]
Just noticed that I was having this issue on `0.7.1` rather than `0.7.0`. Quick checkout and sure enough:

```
2023-03-28T11:51:29.371+0100    INFO    module/state    state/keyring.go:52     constructed keyring signer      {"backend": "test", "path": "/home/jim/.celestia-bridge-blockspacerace-0/keys", "key name": "my_celes_key", "chain-id": "blockspacerace-0"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x22090a1]

goroutine 1 [running]:
github.com/celestiaorg/celestia-node/nodebuilder.(*fsStore).Close(0xc001dbe6f0?)
        /home/jim/celestia-node/nodebuilder/store.go:161 +0x61
go.uber.org/multierr.Invoke.Invoke(0x1?)
        /home/jim/go/pkg/mod/go.uber.org/multierr@v1.9.0/error.go:580 +0x1a
go.uber.org/multierr.AppendInvoke(0xc001d3fa10, {0x35c7240?, 0xc001dbe2b8?})
        /home/jim/go/pkg/mod/go.uber.org/multierr@v1.9.0/error.go:661 +0x30
github.com/celestiaorg/celestia-node/cmd.Start.func1(0xc001a4b200, {0xc00196fb40?, 0x4?, 0x4?})
        /home/jim/celestia-node/cmd/start.go:56 +0x7a2
github.com/spf13/cobra.(*Command).execute(0xc001a4b200, {0xc00196fb00, 0x4, 0x4})
        /home/jim/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:916 +0x862
github.com/spf13/cobra.(*Command).ExecuteC(0x4a6fba0)
        /home/jim/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
        /home/jim/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968
github.com/spf13/cobra.(*Command).ExecuteContext(...)
        /home/jim/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:961
main.run()
        /home/jim/celestia-node/cmd/celestia/main.go:32 +0xab
main.main()
        /home/jim/celestia-node/cmd/celestia/main.go:23 +0x19
```

So a bug in just that version.
